android/app/src/main/java/com/breez/client/plugins/breez/BreezBackup.java

```diff
@@ -0,0 +1,333 @@
+        m_driveResourceClient
+            .getAppFolder()
+            .continueWithTask(task -> {
+                DriveFolder appFolder = task.getResult();
+                SortOrder descendingDateOrder = new SortOrder.Builder().addSortDescending(SortableField.MODIFIED_DATE).build();
+
+                Query query = new Query.Builder()
+                    .addFilter(Filters.eq(SearchableField.MIME_TYPE, DriveFolder.MIME_TYPE))
+                    .addFilter(Filters.eq(SearchableField.TITLE, nodeId))
+                    .setSortOrder(descendingDateOrder)
+                    .build();
+
+                Task<MetadataBuffer> queryTask = m_driveResourceClient.queryChildren(appFolder, query);
+                return  queryTask
+                    .addOnSuccessListener(m_activity,
+                        metadataBuffer -> {
+                            Log.w(TAG, metadataBuffer.toString());
+                            if (metadataBuffer.getCount() == 0) {
+                                result.error("NODE_ID_BACKUP_NOT_FOUND", "Couldn't find node ID backup", null);
+                            }
+
+                            fetchBackupFiles(metadataBuffer.get(0).getDriveId().asDriveFolder(), result);
+                        })
+                    .addOnFailureListener(m_activity, e -> {
+                        Log.e(TAG, "Error retrieving files", e);
+                        result.error("GET_DATA_FAILURE", "Error retrieving files", null);
+                    });
+            });
+    }
+
+    private void fetchBackupFiles(DriveFolder nodeIdFolder, MethodChannel.Result result) {
+        List<String> backupPathsList = new ArrayList<>();
+        Query query = new Query.Builder()
+                .build();
+
+        Task<MetadataBuffer> queryTask = m_driveResourceClient.queryChildren(nodeIdFolder, query);
+
+        queryTask
+            .addOnSuccessListener(m_activity,
+                metadataBuffer -> {
+                    for (Metadata m : metadataBuffer) {
+                        m_driveResourceClient.openFile(m.getDriveId().asDriveFile(), DriveFile.MODE_READ_ONLY)
+                                .addOnSuccessListener(contents -> {
+                            DriveContents fileContents = contents;
+                            InputStream remoteFileInputStream = fileContents.getInputStream();
+
+                            File file = new File(m_activity.getCacheDir().getPath() + "/" + m.getTitle());
+                            FileOutputStream outputStream = null;
+
+                            try {
+                                outputStream = new FileOutputStream(file);
+                            } catch (FileNotFoundException e) {
+                                result.error("CACHE_FILE_FAILURE", "Couldn't find cache file to save to", null);
+                            }
+
+                            try {
+                                int i;
+                                do {
```

Comment:
When the result of "remoteFileInputStream.read(buf)" returns negative we can't write to the outputStream. Here is a suggestion:
```suggestion
                               byte[] buf = new byte[1024];
                               while (i = remoteFileInputStream.read(buf) > 0) {                                  
                                    outputStream.write(buf, 0, i);
                                }
```